### PR TITLE
Update UG and DG

### PR DESCRIPTION
**Pull request by Kurtyjlee**
- Update Ui.png and pdf title
- Add developer guide for assign feature
- Update github templates
- Create sequence diagram for assign groups to events
- Update developer guide use cases
- Updated individual PPP
- Update Find operations in the use cases
- Update UG and DG - UG Documentation for list to be updated to reflect change to list   commands - UG Documentation for add, edit and delete to be updated to add_person,   edit_person, delete_person - UG Documentation for add_person to include new Remark class - DG to be updated for the above as well
- Updated PPP with more information
- Update user guide

### CODE REVIEW

File: docs/UserGuide.md [hunk 1]
```diff
@@ -194,23 +194,46 @@ Expected output when a command succeeds:
 * Output: `Listed all persons`
 * You should see a list of all persons under the Persons column.
 
-### Locating persons by name: `find`
+### Locating persons by name or group: `find_person`
 
-Finds persons whose names contain any of the given keywords.
+Find persons whose names or groups contain any of the given keywords.
 
 Format: `find KEYWORD [MORE_KEYWORDS]`
```
Comment (Ken-Lai) [inline docs/UserGuide.md]: Did you forget to make amendments to the format?

File: docs/UserGuide.md [hunk 2]
```diff
@@ -194,23 +194,46 @@ Expected output when a command succeeds:
...
+
+
+  ![result for 'find alex david'](images/findFriendsResult.png)
+
+
+### Locating events by name, group or person: `find_event`
+
+Find events whose names or groups contain any of the given keywords.
+
+Format: `find KEYWORD [MORE_KEYWORDS]`
```
Comment (Ken-Lai) [inline docs/UserGuide.md]: Same as above

File: docs/UserGuide.md [hunk 3]
```diff
@@ -449,11 +471,14 @@ _Details coming soon ..._
...
 | **Edit Event**   | `edit_event EVENT_INDEX [m/MEETING_DETAILS] [d/DATE] [s/START_TIME] [e/END_TIME] [n/PERSON_NAME]... [u/PERSON_NAME]... [g/GROUP]... [ug/GROUP]...`<br> e.g., `edit_event 1 m/tP week 3 meeting d/2023-10-05 s/1500 e/1700 n/Ken g/CS2103T g/CS2101` |
 | **Delete Event** | `delete_event EVENT_INDEX`<br> e.g., `delete_event 1`                                                                                                                                                                                               |
+| **List Events**   | `list_events`                                                                                                                                                                           |
+| **Find Event**    | `find_event KEYWORD [MORE_KEYWORDS]`<br> e.g., `find meeting`                                                                                                                           |
```
Comment (Ken-Lai) [inline docs/UserGuide.md]: Maybe you forget to include find_all command?

**Review by DistractedCat — APPROVED** [review]
LGTM

**Review by Ken-Lai — COMMENTED** [review]
Minor mistakes in the UG, but overall well done!

**Review by nicleongyj — APPROVED** [review]
LGTM

File: docs/UserGuide.md [hunk 4]
```diff
@@ -172,98 +168,71 @@ Expected output when a command succeeds:
...
 
-Format: `list_persons`
+![DeletePerson](images/Deleteperson.png)
 
-Expected output when a command succeeds:
-* Input: `list_persons`
-* Output: `Listed all persons`
-* You should see a list of all persons under the Persons column.
+Expected output when the command fails:
+* `Invalid command format! delete: Deletes the person identified by the index number used in the displayed person list. Parameters: INDEX (must be a positive integer)Example: delete 1`
```
Comment (nicleongyj) [inline docs/UserGuide.md]: Should change delete to delete_person, as well for other invalid outputs for person commands

**Review by nicleongyj — APPROVED** [review]
Nice

**Pull request merged**